lib/src/middleware_api.dart

```diff
@@ -0,0 +1,30 @@
+library middleware_api;
+
+import 'package:greencat/src/action.dart';
+import 'package:greencat/src/store.dart';
+import 'package:greencat/src/typedefs.dart';
+
+/// Middleware is the suggested way to extend Redux with custom functionality.
+/// Middleware lets you wrap the store’s dispatch method for fun and profit.
+/// The key feature of middleware is that it is composable. Multiple middleware
+/// can be combined together, where each middleware requires no knowledge of
+/// what comes before or after it in the chain.
```

Comment:
It probably is, I ported this as it was, my reasoning was to provide what people are currently familiar with, but certainly favoring patterns that are more idiomatic/natural in dart is better.

Also both approaches could be used in the same application as long as the Store > Reducer are only way to emit new states.

It would be nice to further discuss this offline to make sure I understand your concern :)